### PR TITLE
Feature/issue 44/test dependencies os workflow

### PR DESCRIPTION
**Pull request by Asanto32**
Attempting to add OS, python version, dependency version checks

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -6,18 +6,28 @@ on:
...
     steps:
     - uses: actions/checkout@v4
     - name: Install poetry
       run: pipx install poetry
     - uses: actions/setup-python@v5
       with:
-        python-version-file: pyproject.toml
-        cache: poetry
+        python-version: ${{ matrix.python_version }}
+        cache: pip
```
Comment (nx10) [inline .github/workflows/test.yaml:23]:
```suggestion
        cache: poetry
```

Comment (nx10) [reply .github/workflows/test.yaml:23]: Or does this not work with the version matrix?

Comment (Asanto32) [reply .github/workflows/test.yaml:23]: Copied this from @ReinderVosDeWael implementation haha

Comment (Asanto32) [reply .github/workflows/test.yaml:23]: I think because it uses `uv pip install` it is more efficient to cache pip?

Comment (ReinderVosDeWael) [reply .github/workflows/test.yaml:23]: honestly :shrug: . This is a bit of an edge case for installing that allows us to do the `lowest-direct` and `highest` resolution.

**Review by ReinderVosDeWael — COMMENTED** [review]
I'd suggest changing the on: push event back to only on a push to main. Adding 20-something duplicate test runners to every push that is also a PR is a bit... much.

**Review by ReinderVosDeWael — APPROVED** [review]
